src/DocChecks.jl

```diff
@@ -118,21 +118,34 @@ function __ans__!(m::Module, value)
 end
 
 function doctest(block::Markdown.Code, meta::Dict, doc::Documents.Document, page)
-    m = match(r"jldoctest[ ]?(.*)$", block.language)
-    if m !== nothing
+    lang = block.language
+    if startswith(lang, "jldoctest")
         # Define new module or reuse an old one from this page if we have a named doctest.
-        name = m[1]
+        name = match(r"jldoctest[ ]?(.*)$", split(lang, ';')[1])[1]
         sym = isempty(name) ? gensym("doctest-") : Symbol("doctest-", name)
         sandbox = get!(page.globals.meta, sym) do
             newmod = Module(sym)
             eval(newmod, :(eval(x) = Core.eval($newmod, x)))
             eval(newmod, :(eval(m, x) = Core.eval(m, x)))
             newmod
         end
+        # parse keyword arguments to doctest
+        # kwargstr = split(lang, ';')
+        d = Dict()
+        idx = Compat.findfirst(equalto(';'), lang)
+        if idx !== nothing
+            kwargs = Meta.parse("($(lang[nextind(lang, idx):end]),)")
+            for kwarg in kwargs.args
+                @assert kwarg.head === :(=)
+                @assert isa(kwarg.args[1], Symbol)
```

Comment:
I think it would be good if these errors would be a bit more informative. Since we don't have Markdown line number information, I'd say we should output the filename and the doctest when there's a typo in the arguments.

Reply:
Good idea